Jenkinsfile

```diff
@@ -45,11 +58,12 @@ def rocmtestnode(Map conf) {
             cmake -DCMAKE_C_COMPILER_LAUNCHER=ccache -DCMAKE_CXX_COMPILER_LAUNCHER=ccache -DBUILD_DEV=On -DCMAKE_EXECUTE_PROCESS_COMMAND_ECHO=STDOUT -DMIGRAPHX_DISABLE_VIRTUAL_ENV=ON ${flags} ..
             git diff
             git diff-index --quiet HEAD || (echo "Git repo is not clean after running cmake." && exit 1)
-            make -j\$(nproc) generate VERBOSE=1
+            #make -j\$(nproc) generate VERBOSE=1
             git diff
             git diff-index --quiet HEAD || (echo "Generated files are different. Please run make generate and commit the changes." && exit 1)
-            make -j\$(nproc) all package check VERBOSE=1
+            #make -j\$(nproc) all package check VERBOSE=1
```

Comment:
We should skip the entire stage including docker/cmake builds and even possibly checkout. This should probably be done in the `node(name)` section below, or before if possible since we are just checking a commit hash.

Reply:
Even better would be to do it in(or around) the `Check image` stage. Set an env variable to a list of the contexes that are `success`, and then we can check that variable before we even allocate a node. It would also reduce the number github API calls we need to make(it would be once per build instead of once per build per stage).